IntelliJ 2020.2 PluginException : Psi is outdated

### Short description
May be similar to another question #827 #821

[Download log file](https://drive.google.com/file/d/1hmtpuKMciYmGEUAj7c_Qbz9j9YCgXpxm/view?usp=sharing)

### Version information

 * **IDEA Version:** IntelliJ IDEA 2020.2 Beta  Build #IU-202.6250.13 
 * **JDK Version:** IDE JDK: 11.0.7; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o.  local java version "1.8.0_202"
 * **OS Type & Version:** Windows 10 
 * **Lombok Plugin Version:** 0.30-2020.2EAP
 * **Lombok Dependency Version:** lombok:1.18.8

### Steps to reproduce
_What steps do we need to take to reproduce this issue?_
Oh, I have no good way  reproduce this issue

### Additional information
_Is there any additional information you can provide to help us understanding and solving the issue?_

### Stacktrace
```
2020-07-10 15:20:29,746 [3918732]  ERROR - n.impl.GeneralHighlightingPass - IntelliJ IDEA 2020.2 Beta  Build #IU-202.6250.13 
2020-07-10 15:20:29,746 [3918732]  ERROR - n.impl.GeneralHighlightingPass - JDK: 11.0.7; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2020-07-10 15:20:29,746 [3918732]  ERROR - n.impl.GeneralHighlightingPass - OS: Windows 10 
2020-07-10 15:20:29,747 [3918733]  ERROR - n.impl.GeneralHighlightingPass - Last Action: EditorUp 
2020-07-10 15:20:29,759 [3918745]  ERROR - aemon.impl.PassExecutorService - Element: class de.plushnikov.intellij.plugin.psi.LombokLightMethodBuilder #JAVA  because: psi is outdated
invalidated at: no info [Plugin: Lombook Plugin] 
com.intellij.diagnostic.PluginException: Element: class de.plushnikov.intellij.plugin.psi.LombokLightMethodBuilder #JAVA  because: psi is outdated
invalidated at: no info [Plugin: Lombook Plugin]
	at com.intellij.ide.plugins.PluginManagerCore.createPluginException(PluginManagerCore.java:290)
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:12)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:67)
	at com.intellij.psi.util.PsiUtilCore.ensureValid(PsiUtilCore.java:478)
	at com.intellij.psi.impl.PsiClassImplUtil.lambda$createMembersMap$4(PsiClassImplUtil.java:346)
	at com.intellij.util.containers.ConcurrentFactoryMap$2.create(ConcurrentFactoryMap.java:174)
	at com.intellij.util.containers.ConcurrentFactoryMap.get(ConcurrentFactoryMap.java:40)
	at com.intellij.psi.impl.PsiClassImplUtil$MembersMap.get(PsiClassImplUtil.java:326)
	at com.intellij.psi.impl.PsiClassImplUtil$MembersMap.access$000(PsiClassImplUtil.java:318)
	at com.intellij.psi.impl.PsiClassImplUtil.getMap(PsiClassImplUtil.java:175)
	at com.intellij.psi.impl.PsiClassImplUtil.getAllByMap(PsiClassImplUtil.java:166)
	at com.intellij.psi.impl.PsiClassImplUtil.getAllMethods(PsiClassImplUtil.java:58)
	at com.intellij.psi.impl.source.PsiClassImpl.getAllMethods(PsiClassImpl.java:335)
	at org.jetbrains.plugins.cucumber.java.CucumberJavaUtil.isStepDefinitionClass(CucumberJavaUtil.java:173)
	at org.jetbrains.plugins.cucumber.java.steps.reference.CucumberJavaImplicitUsageProvider.isImplicitUsage(CucumberJavaImplicitUsageProvider.java:15)
	at com.intellij.codeInsight.daemon.impl.UnusedSymbolUtil.isImplicitUsage(UnusedSymbolUtil.java:42)
	at com.intellij.codeInsight.daemon.impl.UnusedSymbolUtil.isReallyUsed(UnusedSymbolUtil.java:300)
	at com.intellij.codeInsight.daemon.impl.UnusedSymbolUtil.isClassUsed(UnusedSymbolUtil.java:289)
	at com.intellij.codeInsight.daemon.impl.analysis.PostHighlightingVisitor.processClass(PostHighlightingVisitor.java:466)
	at com.intellij.codeInsight.daemon.impl.analysis.PostHighlightingVisitor.processIdentifier(PostHighlightingVisitor.java:225)
	at com.intellij.codeInsight.daemon.impl.analysis.PostHighlightingVisitor.collectHighlights(PostHighlightingVisitor.java:144)
	at com.intellij.codeInsight.daemon.impl.analysis.HighlightVisitorImpl.lambda$analyze$1(HighlightVisitorImpl.java:212)
	at com.intellij.codeInsight.daemon.impl.analysis.RefCountHolder.analyze(RefCountHolder.java:362)
	at com.intellij.codeInsight.daemon.impl.analysis.HighlightVisitorImpl.analyze(HighlightVisitorImpl.java:208)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:300)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$analyzeByVisitors$6(GeneralHighlightingPass.java:300)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:96)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:300)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectHighlights(GeneralHighlightingPass.java:268)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:214)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:80)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:54)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:399)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1110)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:392)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:629)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:581)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:391)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:367)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:170)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:182)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:365)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:181)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: com.intellij.psi.PsiInvalidElementAccessException: Element: class de.plushnikov.intellij.plugin.psi.LombokLightMethodBuilder #JAVA  because: psi is outdated
invalidated at: no info
	... 46 more

```

Only workaround I've found is to add new lombok annotations to the source using a different IDE, and then switching back to IntelliJ. When this exception occurs for me, it locks up the entire application.

```2020-07-16 18:55:18,647 [ 218546]  ERROR - ctionSystem.impl.ActionUpdater - IntelliJ IDEA 2020.2 Beta  Build #IU-202.6397.20 
2020-07-16 18:55:18,648 [ 218547]  ERROR - ctionSystem.impl.ActionUpdater - JDK: 11.0.7; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2020-07-16 18:55:18,648 [ 218547]  ERROR - ctionSystem.impl.ActionUpdater - OS: Mac OS X 
2020-07-16 18:55:18,648 [ 218547]  ERROR - ctionSystem.impl.ActionUpdater - Plugin to blame: Lombok version: 0.30-EAP 
2020-07-16 18:55:18,648 [ 218547]  ERROR - ctionSystem.impl.ActionUpdater - Last Action: RunClass 
2020-07-16 18:55:20,868 [ 220767]  ERROR - aemon.impl.PassExecutorService - Element: class de.plushnikov.intellij.plugin.psi.LombokLightMethodBuilder #JAVA  because: psi is outdated
invalidated at: no info [Plugin: Lombook Plugin] 
com.intellij.diagnostic.PluginException: Element: class de.plushnikov.intellij.plugin.psi.LombokLightMethodBuilder #JAVA  because: psi is outdated
invalidated at: no info [Plugin: Lombook Plugin]
	at com.intellij.ide.plugins.PluginManagerCore.createPluginException(PluginManagerCore.java:290)
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:12)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:67)
	at com.intellij.psi.util.PsiUtilCore.ensureValid(PsiUtilCore.java:478)
	at com.intellij.psi.impl.PsiClassImplUtil.lambda$createMembersMap$4(PsiClassImplUtil.java:346)
	at com.intellij.util.containers.ConcurrentFactoryMap$2.create(ConcurrentFactoryMap.java:174)
	at com.intellij.util.containers.ConcurrentFactoryMap.get(ConcurrentFactoryMap.java:40)
	at com.intellij.psi.impl.PsiClassImplUtil$MembersMap.get(PsiClassImplUtil.java:326)
	at com.intellij.psi.impl.PsiClassImplUtil$MembersMap.access$000(PsiClassImplUtil.java:318)
	at com.intellij.psi.impl.PsiClassImplUtil.processCachedMembersByName(PsiClassImplUtil.java:531)
	at com.intellij.psi.impl.PsiClassImplUtil.processDeclarationsInClass(PsiClassImplUtil.java:412)
	at com.intellij.psi.impl.PsiClassImplUtil.processDeclarationsInClass(PsiClassImplUtil.java:389)
	at com.intellij.psi.impl.source.PsiClassImpl.processDeclarations(PsiClassImpl.java:474)
	at com.intellij.psi.scope.util.PsiScopesUtil.treeWalkUp(PsiScopesUtil.java:75)
	at com.intellij.psi.scope.util.PsiScopesUtil.treeWalkUp(PsiScopesUtil.java:54)
	at com.intellij.psi.scope.util.PsiScopesUtil.resolveAndWalk(PsiScopesUtil.java:221)
	at com.intellij.psi.scope.util.PsiScopesUtil.resolveAndWalk(PsiScopesUtil.java:150)
	at com.intellij.psi.impl.source.resolve.PsiResolveHelperImpl.hasOverloads(PsiResolveHelperImpl.java:176)
	at com.intellij.psi.infos.MethodCandidateInfo.lambda$computeWithKnownTargetType$4(MethodCandidateInfo.java:173)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:153)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:54)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$1(CachedValueBase.java:235)
	at com.intellij.openapi.util.RecursionManager$1.doPreventingRecursion(RecursionManager.java:112)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:71)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:236)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:43)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:76)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:150)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:120)
	at com.intellij.psi.infos.MethodCandidateInfo.computeWithKnownTargetType(MethodCandidateInfo.java:168)
	at com.intellij.psi.infos.MethodCandidateInfo.lambda$getPertinentApplicabilityLevelInner$2(MethodCandidateInfo.java:139)
	at com.intellij.openapi.util.RecursionManager$1.doPreventingRecursion(RecursionManager.java:112)
	at com.intellij.psi.infos.MethodCandidateInfo.getPertinentApplicabilityLevelInner(MethodCandidateInfo.java:157)
	at com.intellij.psi.infos.MethodCandidateInfo.getPertinentApplicabilityLevel(MethodCandidateInfo.java:111)
	at com.intellij.psi.scope.conflictResolvers.JavaMethodsConflictResolver.getPertinentApplicabilityLevel(JavaMethodsConflictResolver.java:395)
	at com.intellij.psi.scope.conflictResolvers.JavaMethodsConflictResolver.checkApplicability(JavaMethodsConflictResolver.java:371)
	at com.intellij.psi.scope.conflictResolvers.JavaMethodsConflictResolver.guardedOverloadResolution(JavaMethodsConflictResolver.java:84)
	at com.intellij.psi.scope.conflictResolvers.JavaMethodsConflictResolver.resolveConflict(JavaMethodsConflictResolver.java:55)
	at com.intellij.psi.scope.processor.ConflictFilterProcessor.getResult(ConflictFilterProcessor.java:76)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl.resolveToMethod(PsiReferenceExpressionImpl.java:301)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl.resolve(PsiReferenceExpressionImpl.java:275)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl.access$000(PsiReferenceExpressionImpl.java:45)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl$OurGenericsResolver.resolve(PsiReferenceExpressionImpl.java:183)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl$OurGenericsResolver.resolve(PsiReferenceExpressionImpl.java:173)
	at com.intellij.psi.impl.source.resolve.ResolveCache.lambda$resolveWithCaching$2(ResolveCache.java:185)
	at com.intellij.openapi.util.Computable.get(Computable.java:17)
	at com.intellij.psi.impl.source.resolve.ResolveCache.lambda$loggingResolver$3(ResolveCache.java:227)
	at com.intellij.openapi.util.RecursionManager$1.doPreventingRecursion(RecursionManager.java:112)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:71)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:204)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:185)
	at com.intellij.psi.impl.PsiImplUtil.multiResolveImpl(PsiImplUtil.java:802)
	at com.intellij.psi.impl.PsiImplUtil.multiResolveImpl(PsiImplUtil.java:794)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl.multiResolve(PsiReferenceExpressionImpl.java:338)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl.advancedResolve(PsiReferenceExpressionImpl.java:500)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl.resolve(PsiReferenceExpressionImpl.java:805)
	at com.intellij.codeInsight.TargetElementUtilBase.getReferencedElement(TargetElementUtilBase.java:177)
	at com.intellij.codeInsight.TargetElementUtilBase.doGetReferenceOrReferencedElement(TargetElementUtilBase.java:165)
	at com.intellij.codeInsight.TargetElementUtilBase.getReferenceOrReferencedElement(TargetElementUtilBase.java:204)
	at com.intellij.codeInsight.TargetElementUtilBase.getReferencedElement(TargetElementUtilBase.java:248)
	at com.intellij.codeInsight.TargetElementUtilBase.doFindTargetElement(TargetElementUtilBase.java:227)
	at com.intellij.codeInsight.TargetElementUtilBase.findTargetElement(TargetElementUtilBase.java:295)
	at com.intellij.codeInsight.TargetElementUtil.findTargetElement(TargetElementUtil.java:147)
	at com.intellij.model.psi.impl.TargetsKt.fromTargetEvaluator(targets.kt:166)
	at com.intellij.model.psi.impl.TargetsKt.declarationsOrReferences(targets.kt:127)
	at com.intellij.model.psi.impl.TargetsKt.declaredReferencedData(targets.kt:34)
	at com.intellij.model.psi.impl.TargetsKt.targetSymbols(targets.kt:24)
	at com.intellij.codeInsight.daemon.impl.IdentifierHighlighterPass.getTargetSymbols(IdentifierHighlighterPass.java:233)
	at com.intellij.codeInsight.daemon.impl.IdentifierHighlighterPass.highlightReferencesAndDeclarations(IdentifierHighlighterPass.java:225)
	at com.intellij.codeInsight.daemon.impl.IdentifierHighlighterPass.doCollectInformation(IdentifierHighlighterPass.java:90)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:54)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:399)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1110)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:392)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:629)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:581)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:391)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:367)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:170)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:182)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:365)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:181)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: com.intellij.psi.PsiInvalidElementAccessException: Element: class de.plushnikov.intellij.plugin.psi.LombokLightMethodBuilder #JAVA  because: psi is outdated
invalidated at: no info```

`2020-08-09 22:53:57,345 [2187957]  ERROR - aemon.impl.PassExecutorService - Element: class de.plushnikov.intellij.plugin.psi.LombokLightMethodBuilder #JAVA  because: psi is outdated
invalidated at: no info [Plugin: Lombook Plugin] 
com.intellij.diagnostic.PluginException: Element: class de.plushnikov.intellij.plugin.psi.LombokLightMethodBuilder #JAVA  because: psi is outdated
invalidated at: no info [Plugin: Lombook Plugin]
	at com.intellij.ide.plugins.PluginManagerCore.createPluginException(PluginManagerCore.java:290)
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:12)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:67)
	at com.intellij.psi.util.PsiUtilCore.ensureValid(PsiUtilCore.java:478)
	at com.intellij.psi.impl.PsiClassImplUtil.lambda$createMembersMap$4(PsiClassImplUtil.java:346)
	at com.intellij.util.containers.ConcurrentFactoryMap$2.create(ConcurrentFactoryMap.java:174)
	at com.intellij.util.containers.ConcurrentFactoryMap.get(ConcurrentFactoryMap.java:40)
	at com.intellij.psi.impl.PsiClassImplUtil$MembersMap.get(PsiClassImplUtil.java:326)
	at com.intellij.psi.impl.PsiClassImplUtil$MembersMap.access$000(PsiClassImplUtil.java:318)
	at com.intellij.psi.impl.PsiClassImplUtil.processCachedMembersByName(PsiClassImplUtil.java:531)
	at com.intellij.psi.impl.PsiClassImplUtil.processDeclarationsInClass(PsiClassImplUtil.java:412)
	at com.intellij.psi.impl.PsiClassImplUtil.processDeclarationsInClass(PsiClassImplUtil.java:389)
	at com.intellij.psi.impl.source.PsiClassImpl.processDeclarations(PsiClassImpl.java:474)
	at com.intellij.psi.scope.util.PsiScopesUtil.processQualifierResult(PsiScopesUtil.java:490)
	at com.intellij.psi.scope.util.PsiScopesUtil.processQualifierType(PsiScopesUtil.java:446)
	at com.intellij.psi.scope.util.PsiScopesUtil.setupAndRunProcessor(PsiScopesUtil.java:353)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl.resolveToMethod(PsiReferenceExpressionImpl.java:296)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl.resolve(PsiReferenceExpressionImpl.java:275)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl.access$000(PsiReferenceExpressionImpl.java:45)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl$OurGenericsResolver.resolve(PsiReferenceExpressionImpl.java:183)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl$OurGenericsResolver.resolve(PsiReferenceExpressionImpl.java:173)
	at com.intellij.psi.impl.source.resolve.ResolveCache.lambda$resolveWithCaching$2(ResolveCache.java:185)
	at com.intellij.openapi.util.Computable.get(Computable.java:17)
	at com.intellij.psi.impl.source.resolve.ResolveCache.lambda$loggingResolver$3(ResolveCache.java:227)
	at com.intellij.openapi.util.RecursionManager$1.doPreventingRecursion(RecursionManager.java:112)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:71)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:204)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:185)
	at com.intellij.psi.impl.PsiImplUtil.multiResolveImpl(PsiImplUtil.java:802)
	at com.intellij.psi.impl.PsiImplUtil.multiResolveImpl(PsiImplUtil.java:794)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl.multiResolve(PsiReferenceExpressionImpl.java:338)
	at com.intellij.psi.impl.source.tree.java.PsiReferenceExpressionImpl.advancedResolve(PsiReferenceExpressionImpl.java:500)
	at com.intellij.psi.impl.source.tree.java.PsiMethodCallExpressionImpl.resolveMethodGenerics(PsiMethodCallExpressionImpl.java:52)
	at com.intellij.codeInsight.hints.JavaInlayHintsProvider.hints(JavaHintUtils.kt:65)
	at com.intellij.codeInsight.hints.JavaInlayParameterHintsProvider.getParameterHints(JavaInlayParameterHintsProvider.kt:33)
	at com.intellij.codeInsight.hints.InlayParameterHintsProvider.getParameterHints(InlayParameterHintsProvider.java:29)
	at com.intellij.codeInsight.hints.ParameterHintsPass.process(ParameterHintsPass.java:74)
	at com.intellij.codeInsight.hints.ParameterHintsPass.lambda$doCollectInformation$0(ParameterHintsPass.java:66)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at com.intellij.codeInsight.hints.ParameterHintsPass.doCollectInformation(ParameterHintsPass.java:66)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:54)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:399)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1110)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:392)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:629)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:581)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:391)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:367)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:170)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:182)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:365)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:181)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: com.intellij.psi.PsiInvalidElementAccessException: Element: class de.plushnikov.intellij.plugin.psi.LombokLightMethodBuilder #JAVA  because: psi is outdated
invalidated at: no info
	... 55 more
`

Happens for me on processing @Builder annotation processing. 

It has caused code loss!!!! Because intellij freezes and then all code is lost that is not saved

Facing same issues
When I close the ide and open again the error disapperas

Error:

`com.intellij.diagnostic.PluginException: Element: class de.plushnikov.intellij.plugin.psi.LombokLightMethodBuilder #JAVA  because: psi is outdated
invalidated at: no info [Plugin: Lombook Plugin]
	at com.intellij.ide.plugins.PluginManagerCore.createPluginException(PluginManagerCore.java:290)
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:12)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:67)
	at com.intellij.psi.util.PsiUtilCore.ensureValid(PsiUtilCore.java:478)
	at com.intellij.psi.impl.PsiClassImplUtil.lambda$createMembersMap$4(PsiClassImplUtil.java:346)
	at com.intellij.util.containers.ConcurrentFactoryMap$2.create(ConcurrentFactoryMap.java:174)
	at com.intellij.util.containers.ConcurrentFactoryMap.get(ConcurrentFactoryMap.java:40)
	at com.intellij.psi.impl.PsiClassImplUtil$MembersMap.get(PsiClassImplUtil.java:326)
	at com.intellij.psi.impl.PsiClassImplUtil$MembersMap.access$000(PsiClassImplUtil.java:318)
	at com.intellij.psi.impl.PsiClassImplUtil.getMap(PsiClassImplUtil.java:175)
	at com.intellij.psi.impl.PsiClassImplUtil.getAllByMap(PsiClassImplUtil.java:166)
	at com.intellij.psi.impl.PsiClassImplUtil.getAllMethods(PsiClassImplUtil.java:58)
	at com.intellij.psi.impl.source.PsiClassImpl.getAllMethods(PsiClassImpl.java:335)
	at com.intellij.execution.junit.JUnitUtil.lambda$isJUnit5TestClass$1(JUnitUtil.java:308)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:153)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:54)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$1(CachedValueBase.java:235)
	at com.intellij.openapi.util.RecursionManager$1.doPreventingRecursion(RecursionManager.java:112)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:71)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:236)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:43)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:76)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:150)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:120)
	at com.intellij.execution.junit.JUnitUtil.isJUnit5TestClass(JUnitUtil.java:306)
	at com.intellij.execution.junit.JUnit5Framework.isTestClass(JUnit5Framework.java:56)
	at com.intellij.testIntegration.JavaTestFramework.isTestClass(JavaTestFramework.java:55)
	at com.intellij.codeInsight.TestFrameworksImpl.isTestClass(TestFrameworksImpl.java:18)
	at com.intellij.psi.impl.ElementPresentationUtil.getClassKindImpl(ElementPresentationUtil.java:134)
	at com.intellij.psi.impl.ElementPresentationUtil.lambda$getClassKind$0(ElementPresentationUtil.java:100)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:54)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$1(CachedValueBase.java:235)
	at com.intellij.openapi.util.RecursionManager$1.doPreventingRecursion(RecursionManager.java:112)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:71)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:236)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:43)
	at com.intellij.psi.impl.ElementPresentationUtil.getClassKind(ElementPresentationUtil.java:103)
	at com.intellij.psi.impl.PsiClassImplUtil.lambda$static$0(PsiClassImplUtil.java:226)
	at com.intellij.ui.IconDeferrerImpl.deferImpl(IconDeferrerImpl.java:70)
	at com.intellij.ui.IconDeferrerImpl.defer(IconDeferrerImpl.java:60)
	at com.intellij.psi.impl.PsiClassImplUtil.getClassIcon(PsiClassImplUtil.java:248)
	at com.intellij.psi.impl.PsiClassImplUtil.getClassIcon(PsiClassImplUtil.java:235)
	at com.intellij.psi.impl.source.PsiClassImpl.getElementIcon(PsiClassImpl.java:533)
	at com.intellij.psi.impl.ElementBase.doComputeIconNow(ElementBase.java:110)
	at com.intellij.psi.impl.ElementBase.lambda$computeIconNow$2(ElementBase.java:97)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:126)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:115)
	at com.intellij.psi.impl.ElementBase.computeIconNow(ElementBase.java:97)
	at com.intellij.psi.impl.ElementBase.lambda$static$0(ElementBase.java:41)
	at com.intellij.ui.CoreIconManager.lambda$createDeferredIcon$0(CoreIconManager.java:48)
	at com.intellij.ui.IconDeferrerImpl.deferImpl(IconDeferrerImpl.java:70)
	at com.intellij.ui.IconDeferrerImpl.defer(IconDeferrerImpl.java:60)
	at com.intellij.ui.CoreIconManager.createDeferredIcon(CoreIconManager.java:48)
	at com.intellij.psi.impl.ElementBase.computeIcon(ElementBase.java:89)
	at com.intellij.psi.impl.ElementBase.getIcon(ElementBase.java:68)
	at com.intellij.ide.JavaFileIconPatcher.patchIcon(JavaFileIconPatcher.java:56)
	at com.intellij.util.IconUtil.lambda$static$0(IconUtil.java:155)
	at com.intellij.ui.DeferredIconImpl.evaluate(DeferredIconImpl.java:279)
	at com.intellij.ui.DeferredIconImpl.lambda$scheduleEvaluation$0(DeferredIconImpl.java:166)
	at com.intellij.ui.IconDeferrerImpl.evaluateDeferred(IconDeferrerImpl.java:98)
	at com.intellij.ui.DeferredIconImpl.lambda$scheduleEvaluation$1(DeferredIconImpl.java:166)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1110)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:77)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:153)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:114)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:56)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:170)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:629)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:581)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:56)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:111)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:77)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:98)
	at com.intellij.ui.DeferredIconImpl.lambda$scheduleEvaluation$4(DeferredIconImpl.java:165)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:215)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:26)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:194)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:207)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:183)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.intellij.psi.PsiInvalidElementAccessException: Element: class de.plushnikov.intellij.plugin.psi.LombokLightMethodBuilder #JAVA  because: psi is outdated
invalidated at: no info
	... 87 more`

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightClassBuilder.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightClassBuilder.java
@@ -1,6 +1,5 @@
 package de.plushnikov.intellij.plugin.psi;
 
-import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.psi.PsiClass;
 import com.intellij.psi.PsiClassType;
@@ -28,8 +27,6 @@ import java.util.List;
 import java.util.stream.Stream;
 
 public class LombokLightClassBuilder extends LightPsiClassBuilder implements PsiExtensibleClass {
-  private static final Logger LOG = Logger.getInstance(LombokLightClassBuilder.class);
-
   private boolean myIsEnum;
   private final String myQualifiedName;
   private final Icon myBaseIcon;
@@ -203,9 +200,6 @@ public class LombokLightClassBuilder extends LightPsiClassBuilder implements Psi
     LombokLightClassBuilder that = (LombokLightClassBuilder) o;
 
     if(getNavigationElement() != this && !getNavigationElement().equals(that.getNavigationElement())) {
-      if(myQualifiedName.equals(that.myQualifiedName)) {
-        LOG.warn("Usually I would have been equal!");
-      }
       return false;
     }
 

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightClassBuilder.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightClassBuilder.java
@@ -1,5 +1,6 @@
 package de.plushnikov.intellij.plugin.psi;
 
+import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.psi.PsiClass;
 import com.intellij.psi.PsiClassType;
@@ -27,6 +28,8 @@ import java.util.List;
 import java.util.stream.Stream;
 
 public class LombokLightClassBuilder extends LightPsiClassBuilder implements PsiExtensibleClass {
+  private static final Logger LOG = Logger.getInstance(LombokLightClassBuilder.class);
+
   private boolean myIsEnum;
   private final String myQualifiedName;
   private final Icon myBaseIcon;
@@ -198,6 +201,13 @@ public class LombokLightClassBuilder extends LightPsiClassBuilder implements Psi
     }
 
     LombokLightClassBuilder that = (LombokLightClassBuilder) o;
+
+    if(getNavigationElement() != this && !getNavigationElement().equals(that.getNavigationElement())) {
+      if(myQualifiedName.equals(that.myQualifiedName)) {
+        LOG.warn("Usually I would have been equal!");
+      }
+      return false;
+    }
 
     return myQualifiedName.equals(that.myQualifiedName);
   }

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightFieldBuilder.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightFieldBuilder.java
@@ -1,6 +1,7 @@
 package de.plushnikov.intellij.plugin.psi;
 
 import com.intellij.lang.java.JavaLanguage;
+import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.psi.PsiClass;
 import com.intellij.psi.PsiElement;
@@ -23,6 +24,8 @@ import java.util.stream.Stream;
  * @author Plushnikov Michail
  */
 public class LombokLightFieldBuilder extends LightFieldBuilder {
+  private static final Logger LOG = Logger.getInstance(LombokLightFieldBuilder.class);
+
   private String myName;
   private final LombokLightIdentifier myNameIdentifier;
   private final LombokLightModifierList myModifierList;
@@ -141,6 +144,13 @@ public class LombokLightFieldBuilder extends LightFieldBuilder {
 
         stillEquivalent = (null == containingClass && null == anotherContainingClass) ||
           (null != containingClass && containingClass.isEquivalentTo(anotherContainingClass));
+      }
+
+      if(getNavigationElement() != this && !getNavigationElement().equals(anotherLightField.getNavigationElement())) {
+        if(stillEquivalent) {
+          LOG.warn("Usually I would have been equal!");
+        }
+        stillEquivalent = false;
       }
 
       return stillEquivalent;

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightFieldBuilder.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightFieldBuilder.java
@@ -1,7 +1,6 @@
 package de.plushnikov.intellij.plugin.psi;
 
 import com.intellij.lang.java.JavaLanguage;
-import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.psi.PsiClass;
 import com.intellij.psi.PsiElement;
@@ -24,8 +23,6 @@ import java.util.stream.Stream;
  * @author Plushnikov Michail
  */
 public class LombokLightFieldBuilder extends LightFieldBuilder {
-  private static final Logger LOG = Logger.getInstance(LombokLightFieldBuilder.class);
-
   private String myName;
   private final LombokLightIdentifier myNameIdentifier;
   private final LombokLightModifierList myModifierList;
@@ -147,9 +144,6 @@ public class LombokLightFieldBuilder extends LightFieldBuilder {
       }
 
       if(getNavigationElement() != this && !getNavigationElement().equals(anotherLightField.getNavigationElement())) {
-        if(stillEquivalent) {
-          LOG.warn("Usually I would have been equal!");
-        }
         stillEquivalent = false;
       }
 

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightIdentifier.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightIdentifier.java
@@ -1,14 +1,19 @@
 package de.plushnikov.intellij.plugin.psi;
 
+import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.psi.PsiElement;
 import com.intellij.psi.PsiManager;
 import com.intellij.psi.impl.light.LightIdentifier;
 
+import java.util.Objects;
+
 /**
  * Date: 12.10.13 Time: 23:27
  */
 public class LombokLightIdentifier extends LightIdentifier {
+  private static final Logger LOG = Logger.getInstance(LombokLightIdentifier.class);
+
   private String myText;
 
   public LombokLightIdentifier(PsiManager manager, String text) {
@@ -47,7 +52,14 @@ public class LombokLightIdentifier extends LightIdentifier {
 
     LombokLightIdentifier that = (LombokLightIdentifier) o;
 
-    return !(myText != null ? !myText.equals(that.myText) : that.myText != null);
+    if(getNavigationElement() != this && !getNavigationElement().equals(that.getNavigationElement())) {
+      if(Objects.equals(myText, that.myText)) {
+        LOG.warn("Usually I would have been equal!");
+      }
+      return false;
+    }
+
+    return Objects.equals(myText, that.myText);
 
   }
 

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightIdentifier.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightIdentifier.java
@@ -1,6 +1,5 @@
 package de.plushnikov.intellij.plugin.psi;
 
-import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.psi.PsiElement;
 import com.intellij.psi.PsiManager;
@@ -12,8 +11,6 @@ import java.util.Objects;
  * Date: 12.10.13 Time: 23:27
  */
 public class LombokLightIdentifier extends LightIdentifier {
-  private static final Logger LOG = Logger.getInstance(LombokLightIdentifier.class);
-
   private String myText;
 
   public LombokLightIdentifier(PsiManager manager, String text) {
@@ -53,9 +50,6 @@ public class LombokLightIdentifier extends LightIdentifier {
     LombokLightIdentifier that = (LombokLightIdentifier) o;
 
     if(getNavigationElement() != this && !getNavigationElement().equals(that.getNavigationElement())) {
-      if(Objects.equals(myText, that.myText)) {
-        LOG.warn("Usually I would have been equal!");
-      }
       return false;
     }
 

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightMethodBuilder.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightMethodBuilder.java
@@ -2,6 +2,7 @@ package de.plushnikov.intellij.plugin.psi;
 
 import com.intellij.lang.ASTNode;
 import com.intellij.lang.java.JavaLanguage;
+import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.psi.*;
 import com.intellij.psi.impl.CheckUtil;
@@ -23,6 +24,8 @@ import java.util.Objects;
  * @author Plushnikov Michail
  */
 public class LombokLightMethodBuilder extends LightMethodBuilder {
+  private static final Logger LOG = Logger.getInstance(LombokLightMethodBuilder.class);
+
   private PsiMethod myMethod;
   private ASTNode myASTNode;
   private PsiCodeBlock myBodyCodeBlock;
@@ -287,6 +290,14 @@ public class LombokLightMethodBuilder extends LightMethodBuilder {
     if (!getParameterList().equals(that.getParameterList())) {
       return false;
     }
+
+    if(getNavigationElement() != this && !getNavigationElement().equals(that.getNavigationElement())) {
+      if(Objects.equals(myReturnTypeAsText, that.myReturnTypeAsText)) {
+        LOG.warn("Usually I would have been equal!");
+      }
+      return false;
+    }
+
     return Objects.equals(myReturnTypeAsText, that.myReturnTypeAsText);
   }
 

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightMethodBuilder.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightMethodBuilder.java
@@ -2,7 +2,6 @@ package de.plushnikov.intellij.plugin.psi;
 
 import com.intellij.lang.ASTNode;
 import com.intellij.lang.java.JavaLanguage;
-import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.psi.*;
 import com.intellij.psi.impl.CheckUtil;
@@ -24,8 +23,6 @@ import java.util.Objects;
  * @author Plushnikov Michail
  */
 public class LombokLightMethodBuilder extends LightMethodBuilder {
-  private static final Logger LOG = Logger.getInstance(LombokLightMethodBuilder.class);
-
   private PsiMethod myMethod;
   private ASTNode myASTNode;
   private PsiCodeBlock myBodyCodeBlock;
@@ -292,9 +289,6 @@ public class LombokLightMethodBuilder extends LightMethodBuilder {
     }
 
     if(getNavigationElement() != this && !getNavigationElement().equals(that.getNavigationElement())) {
-      if(Objects.equals(myReturnTypeAsText, that.myReturnTypeAsText)) {
-        LOG.warn("Usually I would have been equal!");
-      }
       return false;
     }
 

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightModifierList.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightModifierList.java
@@ -1,6 +1,7 @@
 package de.plushnikov.intellij.plugin.psi;
 
 import com.intellij.lang.Language;
+import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.psi.JavaPsiFacade;
 import com.intellij.psi.PsiAnnotation;
@@ -23,6 +24,7 @@ import java.util.Set;
  * @author Plushnikov Michail
  */
 public class LombokLightModifierList extends LightModifierList {
+  private static final Logger LOG = Logger.getInstance(LombokLightModifierList.class);
   private static final Set<String> ALL_MODIFIERS = new HashSet<>(Arrays.asList(PsiModifier.MODIFIERS));
 
   private final Map<String, PsiAnnotation> myAnnotations;
@@ -123,6 +125,13 @@ public class LombokLightModifierList extends LightModifierList {
     }
 
     LombokLightModifierList that = (LombokLightModifierList) o;
+
+    if (getNavigationElement() != this && !getNavigationElement().equals(that.getNavigationElement())) {
+      if(myAnnotations.equals(that.myAnnotations)) {
+        LOG.warn("Usually I would have been equal!");
+      }
+      return false;
+    }
 
     return myAnnotations.equals(that.myAnnotations);
   }

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightModifierList.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightModifierList.java
@@ -1,7 +1,6 @@
 package de.plushnikov.intellij.plugin.psi;
 
 import com.intellij.lang.Language;
-import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.psi.JavaPsiFacade;
 import com.intellij.psi.PsiAnnotation;
@@ -24,7 +23,6 @@ import java.util.Set;
  * @author Plushnikov Michail
  */
 public class LombokLightModifierList extends LightModifierList {
-  private static final Logger LOG = Logger.getInstance(LombokLightModifierList.class);
   private static final Set<String> ALL_MODIFIERS = new HashSet<>(Arrays.asList(PsiModifier.MODIFIERS));
 
   private final Map<String, PsiAnnotation> myAnnotations;
@@ -127,9 +125,6 @@ public class LombokLightModifierList extends LightModifierList {
     LombokLightModifierList that = (LombokLightModifierList) o;
 
     if (getNavigationElement() != this && !getNavigationElement().equals(that.getNavigationElement())) {
-      if(myAnnotations.equals(that.myAnnotations)) {
-        LOG.warn("Usually I would have been equal!");
-      }
       return false;
     }
 

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightParameter.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightParameter.java
@@ -1,7 +1,6 @@
 package de.plushnikov.intellij.plugin.psi;
 
 import com.intellij.lang.Language;
-import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.psi.PsiElement;
 import com.intellij.psi.PsiIdentifier;
@@ -19,8 +18,6 @@ import java.util.Collections;
  * @author Plushnikov Michail
  */
 public class LombokLightParameter extends LightParameter {
-  private static final Logger LOG = Logger.getInstance(LombokLightParameter.class);
-
   private String myName;
   private final LombokLightIdentifier myNameIdentifier;
 
@@ -81,9 +78,6 @@ public class LombokLightParameter extends LightParameter {
     }
 
     if(getNavigationElement() != this && !getNavigationElement().equals(that.getNavigationElement())) {
-      if(thisType.getCanonicalText().equals(thatType.getCanonicalText())) {
-        LOG.warn("Usually I would have been equal!");
-      }
       return false;
     }
 

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightParameter.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightParameter.java
@@ -1,6 +1,7 @@
 package de.plushnikov.intellij.plugin.psi;
 
 import com.intellij.lang.Language;
+import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.psi.PsiElement;
 import com.intellij.psi.PsiIdentifier;
@@ -18,6 +19,8 @@ import java.util.Collections;
  * @author Plushnikov Michail
  */
 public class LombokLightParameter extends LightParameter {
+  private static final Logger LOG = Logger.getInstance(LombokLightParameter.class);
+
   private String myName;
   private final LombokLightIdentifier myNameIdentifier;
 
@@ -74,6 +77,13 @@ public class LombokLightParameter extends LightParameter {
     final PsiType thisType = getType();
     final PsiType thatType = that.getType();
     if (thisType.isValid() != thatType.isValid()) {
+      return false;
+    }
+
+    if(getNavigationElement() != this && !getNavigationElement().equals(that.getNavigationElement())) {
+      if(thisType.getCanonicalText().equals(thatType.getCanonicalText())) {
+        LOG.warn("Usually I would have been equal!");
+      }
       return false;
     }
 

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightParameterListBuilder.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightParameterListBuilder.java
@@ -1,12 +1,15 @@
 package de.plushnikov.intellij.plugin.psi;
 
 import com.intellij.lang.Language;
+import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.psi.PsiManager;
 import com.intellij.psi.impl.light.LightParameterListBuilder;
 
 import java.util.Arrays;
 
 public class LombokLightParameterListBuilder extends LightParameterListBuilder {
+  private static final Logger LOG = Logger.getInstance(LombokLightParameterListBuilder.class);
+
   public LombokLightParameterListBuilder(PsiManager manager, Language language) {
     super(manager, language);
   }
@@ -23,6 +26,13 @@ public class LombokLightParameterListBuilder extends LightParameterListBuilder {
     LombokLightParameterListBuilder that = (LombokLightParameterListBuilder) o;
 
     if (getParametersCount() != that.getParametersCount()) {
+      return false;
+    }
+
+    if(getNavigationElement() != this && !getNavigationElement().equals(that.getNavigationElement())) {
+      if(Arrays.equals(getParameters(), that.getParameters())) {
+        LOG.warn("Usually I would have been equal!");
+      }
       return false;
     }
 

--- a/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightParameterListBuilder.java
+++ b/src/main/java/de/plushnikov/intellij/plugin/psi/LombokLightParameterListBuilder.java
@@ -1,14 +1,12 @@
 package de.plushnikov.intellij.plugin.psi;
 
 import com.intellij.lang.Language;
-import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.psi.PsiManager;
 import com.intellij.psi.impl.light.LightParameterListBuilder;
 
 import java.util.Arrays;
 
 public class LombokLightParameterListBuilder extends LightParameterListBuilder {
-  private static final Logger LOG = Logger.getInstance(LombokLightParameterListBuilder.class);
 
   public LombokLightParameterListBuilder(PsiManager manager, Language language) {
     super(manager, language);
@@ -30,9 +28,6 @@ public class LombokLightParameterListBuilder extends LightParameterListBuilder {
     }
 
     if(getNavigationElement() != this && !getNavigationElement().equals(that.getNavigationElement())) {
-      if(Arrays.equals(getParameters(), that.getParameters())) {
-        LOG.warn("Usually I would have been equal!");
-      }
       return false;
     }
 

